tests/unit/db.js

```diff
@@ -39,6 +47,47 @@ function checkValueNotInDb(db, k, done) {
     });
 }
 
+function checkKeyInDB(db, key, cb) {
+    return db.get(key, (err, value) => {
+        if (value || (err && !err.stack.includes('Error: NotFound:'))) {
+            return cb(errors.EntityAlreadyExists);
+        }
+        return cb();
+    });
+}
+
+class ConditionalLevelDB {
+    constructor() {
+        this.db = createDb();
+    }
+
+    batch(conditionalOperations, writeOptions, cb) {
+        const { db } = this.db;
+        const operations = Array.from(conditionalOperations);
+        return async.eachOfLimit(operations, 10, (op, index, asyncCallback) => {
+            if (!op.notExists) {
+                return asyncCallback(null, true);
+            }
+            return async.series([
+                next => checkKeyInDB(db, op.notExists, next),
+                next => {
+                    operations.splice(index, 1);
+                    return next();
+                },
+            ], asyncCallback);
+        }, err => {
+            if (err) {
+                return cb(err);
+            }
+            return this.db.batch(operations, writeOptions, cb);
```

Comment:
It should then be ok to pass the `this.conditions` array inside the `writeOptions`, e.g. in `writeOptions.conditions`.